VideoLocker/src/main/java/org/edx/mobile/view/view_holders/AuthorLayoutViewHolder.java

```diff
@@ -1,16 +1,110 @@
+
+    public final ViewGroup profileRow;
+    public final ImageView profileImageView;
+    public final TextView authorTextView;
+    public final TextView dateTextView;
+    public final TextView answerTextView;
 
     public AuthorLayoutViewHolder(View itemView) {
-        discussionAuthorTextView = (TextView) itemView.
-                findViewById(R.id.discussion_author_layout_author_text_view);
+        profileRow = (ViewGroup) itemView;
+        profileImageView = (ImageView) itemView.findViewById(R.id.profile_image);
+        authorTextView = (TextView) itemView.findViewById(R.id.discussion_author_text_view);
+        dateTextView = (TextView) itemView.findViewById(R.id.discussion_date_text_view);
+        answerTextView = (TextView) itemView.findViewById(R.id.discussion_responses_answer_text_view);
+
+        final Context context = answerTextView.getContext();
+        TextViewCompat.setCompoundDrawablesRelativeWithIntrinsicBounds(
+                answerTextView,
+                new IconDrawable(context, FontAwesomeIcons.fa_check_square_o)
+                        .sizeRes(context, R.dimen.edx_xxx_small)
+                        .colorRes(context, R.color.edx_utility_success),
+                null, null, null);
+        RoboGuice.getInjector(context).injectMembers(this);
+    }
+
+    public void populateViewHolder(@NonNull Config config, @NonNull IAuthorData authorData,
+                                   @NonNull ProfileImageProvider provider,
+                                   long initialTimeStampMs,
+                                   @NonNull final Runnable listener) {
+        final Context context = profileImageView.getContext();
+        final ProfileImage profileImage;
+        {
+            if (provider.getProfileImage() != null) {
+                profileImage = provider.getProfileImage();
+            } else {
+                // TODO: Remove this else block when MA-2542 is fixed.
+                /**
+                 * Background: Currently the POST & PUT APIs aren't configured to return a user's
+                 * {@link ProfileImage} in their response. Since, the currently logged-in user is
+                 * the only one that can POST using the app, so, we use the locally stored
+                 * {@link ProfileImage} in {@link UserPrefs} instead.
+                 * Incase of PUT we just use the previous {@link ProfileImage} version that we had.
+                 */
+                ProfileModel profileModel = userPrefs.getProfile();
+                if (profileModel != null && authorData.getAuthor().equals(profileModel.username)) {
+                    profileImage = userPrefs.getProfileImage();
+                } else {
+                    profileImage = null;
+                }
+            }
+            if (profileImage != null && profileImage.hasImage()) {
+                Glide.with(context).load(profileImage.getImageUrlMedium())
+                        .into(profileImageView);
```

Comment:
The fallback image should be declared in this case as well, to be shown while Glide is loading the actual image from the server.

Also, should there be a concurrent thumbnail request for the small version of the image while the large is loading? Glide supports loading thumbnails in this manner.